--- glib-2.15.0/glib/gtestutils.h.jj	2007-12-21 05:53:51.000000000 +0100
+++ glib-2.15.0/glib/gtestutils.h	2007-12-21 11:50:46.000000000 +0100
@@ -22,7 +22,7 @@
 
 #include <glib.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 typedef struct GTestCase  GTestCase;
 typedef struct GTestSuite GTestSuite;
@@ -118,7 +118,7 @@ void    g_test_queue_destroy            
 typedef enum {
   G_TEST_TRAP_SILENCE_STDOUT    = 1 << 7,
   G_TEST_TRAP_SILENCE_STDERR    = 1 << 8,
-  G_TEST_TRAP_INHERIT_STDIN     = 1 << 9,
+  G_TEST_TRAP_INHERIT_STDIN     = 1 << 9
 } GTestTrapFlags;
 gboolean g_test_trap_fork               (guint64              usec_timeout,
                                          GTestTrapFlags       test_trap_flags);
@@ -215,7 +215,7 @@ typedef enum {
   G_TEST_LOG_STOP_CASE,         // d:status d:nforks d:elapsed
   G_TEST_LOG_MIN_RESULT,        // s:blurb d:result
   G_TEST_LOG_MAX_RESULT,        // s:blurb d:result
-  G_TEST_LOG_MESSAGE,           // s:blurb
+  G_TEST_LOG_MESSAGE            // s:blurb
 } GTestLogType;
 
 typedef struct {
@@ -240,6 +240,6 @@ void            g_test_log_buffer_push  
 GTestLogMsg*    g_test_log_buffer_pop   (GTestLogBuffer *tbuffer);
 void            g_test_log_msg_free     (GTestLogMsg    *tmsg);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* __G_TEST_UTILS_H__ */
