diff -up glib-2.22.0/configure.in.pyloc glib-2.22.0/configure.in
--- glib-2.22.0/configure.in.pyloc	2009-09-22 09:40:56.000000000 -0400
+++ glib-2.22.0/configure.in	2009-09-23 15:37:23.729418282 -0400
@@ -256,8 +256,10 @@ AC_ARG_WITH(runtime-libdir,
                            [Install runtime libraries relative to libdir])],
            [],
 	   [with_runtime_libdir=""])
-GLIB_RUNTIME_LIBDIR=$with_runtime_libdir
+GLIB_RUNTIME_LIBDIR="$with_runtime_libdir"
+ABS_GLIB_RUNTIME_LIBDIR="`readlink -f $libdir/$with_runtime_libdir`"
 AC_SUBST(GLIB_RUNTIME_LIBDIR)
+AC_SUBST(ABS_GLIB_RUNTIME_LIBDIR)
 AM_CONDITIONAL(HAVE_GLIB_RUNTIME_LIBDIR, [test "x$with_runtime_libdir" != "x"])
 
 dnl Checks for programs.
diff -up glib-2.22.0/glib/Makefile.am.pyloc glib-2.22.0/glib/Makefile.am
--- glib-2.22.0/glib/Makefile.am.pyloc	2009-09-22 16:19:35.000000000 -0400
+++ glib-2.22.0/glib/Makefile.am	2009-09-23 15:53:09.109395306 -0400
@@ -380,8 +380,8 @@ libglib-gdb.py: libglib-gdb.py.in
 
 
 install-data-hook: libglib-gdb.py
-	mkdir -p $(DESTDIR)$(datadir)/gdb/auto-load${libdir}
-	$(INSTALL) libglib-gdb.py $(DESTDIR)$(datadir)/gdb/auto-load${libdir}/libglib-2.0.so.0.$(LT_CURRENT).0-gdb.py
+	mkdir -p $(DESTDIR)$(datadir)/gdb/auto-load$(ABS_GLIB_RUNTIME_LIBDIR)
+	$(INSTALL) libglib-gdb.py $(DESTDIR)$(datadir)/gdb/auto-load$(ABS_GLIB_RUNTIME_LIBDIR)/libglib-2.0.so.0.$(LT_CURRENT).0-gdb.py
 if HAVE_GLIB_RUNTIME_LIBDIR
 	mkdir -p $(DESTDIR)$(libdir)/$(GLIB_RUNTIME_LIBDIR)
 	mv $(DESTDIR)$(libdir)/libglib-2.0.so.0 $(DESTDIR)$(libdir)/$(GLIB_RUNTIME_LIBDIR)
diff -up glib-2.22.0/gobject/Makefile.am.pyloc glib-2.22.0/gobject/Makefile.am
--- glib-2.22.0/gobject/Makefile.am.pyloc	2009-09-22 16:19:20.000000000 -0400
+++ glib-2.22.0/gobject/Makefile.am	2009-09-23 15:53:18.011645753 -0400
@@ -274,8 +274,8 @@ uninstall-gdb:
 	-rm -r $(DESTDIR)$(datadir)/gdb
 
 install-data-hook: libgobject-gdb.py
-	mkdir -p $(DESTDIR)$(datadir)/gdb/auto-load${libdir}
-	$(INSTALL) libgobject-gdb.py $(DESTDIR)$(datadir)/gdb/auto-load${libdir}/libgobject-2.0.so.0.$(LT_CURRENT).0-gdb.py
+	mkdir -p $(DESTDIR)$(datadir)/gdb/auto-load/$(ABS_GLIB_RUNTIME_LIBDIR)
+	$(INSTALL) libgobject-gdb.py $(DESTDIR)$(datadir)/gdb/auto-load/$(ABS_GLIB_RUNTIME_LIBDIR)/libgobject-2.0.so.0.$(LT_CURRENT).0-gdb.py
 if HAVE_GLIB_RUNTIME_LIBDIR
 	mkdir -p $(DESTDIR)$(libdir)/$(GLIB_RUNTIME_LIBDIR)
 	mv $(DESTDIR)$(libdir)/libgobject-2.0.so.0 $(DESTDIR)$(libdir)/$(GLIB_RUNTIME_LIBDIR)
